lib/pure/strutils.nim

```diff
@@ -1036,14 +1038,32 @@ func toHex*(s: string): string {.rtl.} =
     doAssert b.toHex() == "41"
     doAssert c.toHex() == "00FF"
 
-  const HexChars = "0123456789ABCDEF"
   result = newString(s.len * 2)
   for pos, c in s:
     var n = ord(c)
     result[pos * 2 + 1] = HexChars[n and 0xF]
     n = n shr 4
     result[pos * 2] = HexChars[n]
 
+func toHex*(data: openarray[byte]): string {.rtl.} =
```

Comment:
As I said, it belongs to parseutils and should probably store the result into a `var openArray[char]`. If you avoid strings, avoid them properly.